tests/unit/lib/deploy/test_deployer.py

```diff
@@ -90,6 +91,11 @@ def test_deployer_init_default_sleep(self):
         deployer = Deployer(MagicMock().client("cloudformation"))
         self.assertEqual(deployer.client_sleep, 0.5)
 
+    @patch("os.environ", {**os.environ, "SAM_CLI_POLL_DELAY": 30})
+    def test_deployer_sam_cli_poll_delay(self):
+        deployer = Deployer(MagicMock().client("cloudformation"), client_sleep=os.getenv("SAM_CLI_POLL_DELAY", 0.5))
+        self.assertEqual(deployer.client_sleep, 30)
+
```

Comment:
@lucashuy makes sense. I removed it from here and made a minor change in command unit to make sure we are capturing the os env variable since it already captures the tests related to `SAM_CLI_POLL_DELAY`